recipes/libjpeg-turbo12/meta.yaml

```diff
@@ -0,0 +1,38 @@
+{% set version = "2.0.0" %}
```

Comment:
This is already packaged, but is called [`libjpeg-turbo`]( https://github.com/conda-forge/libjpeg-turbo-feedstock ). Would just use that if possible. If there are issues using it, would raise with the maintainers at that feedstock.

Reply:
Please see comment https://github.com/conda-forge/staged-recipes/pull/6842#issuecomment-431105910 .
It is a version with different compile-time settings. It would be possible to add it on top of normal `libjpeg-turbo`, but this would roughly double package size for everyone, while being useless to most users.

Reply:
@csachs how does it work mixing both 8bit and 12bit turbo; what about mixing turbo and libjpeg?

Experience tells that we should be very careful. In general, it is not enough that we avoid file clobbering, as we do here. The hard part is to avoid symbol clashing with libjpeg, which unfortunately is current default in conda-forge. Also I wonder about mixing libjpeg-turbo 8bit and 12bit (see upstream issue at https://github.com/libjpeg-turbo/libjpeg-turbo/issues/199). Depending on the order libraries are loaded by the dynamic linker, we can end up in hell.

I think a libjpeg-turbo12 package would be a fine thing to have if symbols would be properly prefixed, for linking statically or if it becomes supported by libjpeg-turbo itself. Otherwise perhaps we have a bit our hands tied, since I think at the moment 8bit and 12bit turbo are not safe to parallel install.

For reference, I have a test for libjpeg vs turbo we could maybe adapt: https://github.com/loopbio/opencv-feedstock/blob/loopbio/recipe/ensure_jpegturbo_opencv_plays_nicely_with_jpeg9.py (but note that opencv is a little complex beast because of how it re-exports symbols). I also have a little nasty turbo symbol renamer here: https://github.com/loopbio/libjpeg-turbo-feedstock.

Reply:
Is this such a big problem? I have not encountered any segfaults with this yet; @cgohlke has one Python `.so` loading the 8 bit bindings, one `.so` loading the 12 bit bindings.
The 12 bit libjpeg-turbo12 variant is imo definetly safe to install along the 8 bit variant, as it uses completely different paths:
```
bin/cjpeg12
bin/djpeg12
bin/jpegtran12
bin/rdjpgcom12
bin/wrjpgcom12
include/jpeg12/jconfig.h
include/jpeg12/jerror.h
include/jpeg12/jmorecfg.h
include/jpeg12/jpeglib.h
lib/libjpeg12.a
lib/libjpeg12.so
lib/libjpeg12.so.62
lib/libjpeg12.so.62.3.0
lib/pkgconfig/libjpeg12.pc
lib/pkgconfig/libturbojpeg12.pc
share/doc/libjpeg-turbo12/example.txt
share/doc/libjpeg-turbo12/libjpeg.txt
share/doc/libjpeg-turbo12/LICENSE.md
share/doc/libjpeg-turbo12/README.ijg
share/doc/libjpeg-turbo12/README.md
share/doc/libjpeg-turbo12/structure.txt
share/doc/libjpeg-turbo12/tjexample.c
share/doc/libjpeg-turbo12/usage.txt
share/doc/libjpeg-turbo12/wizard.txt
share/man/man1/cjpeg12.1
share/man/man1/djpeg12.1
share/man/man1/jpegtran12.1
share/man/man1/rdjpgcom12.1
share/man/man1/wrjpgcom12.1

```
Someone who does not link it or does not use this (`imagecodecs`) module, should never unintentionally come into contact with it.
Loading `imagecodecs` and OpenCV or PIL (in various orders) works without problems for me (using the script you mentioned running the decoding example).
The `imagecodecs`-provided test suite works for me for 8 bit and 12 bit JPEGs, I used `libjpeg-turbo`, as this is what @cgohlke mentioned in his README.

Reply:
It is not about the file paths. It is about the symbol names, that are the same in jpeg 9 and libjpeg-turbo (8bits and 12bits).

Could you please add these tests to the recipe?

Reply:
The tests of `imagecodecs` require some other packages not currently in conda-forge… but I can add them nonetheless. I expect a certain amount of skips.